GWXtreme/parametrized_eos_sampler.py

```diff
@@ -179,7 +179,61 @@ def save_data(self):
         f.create_dataset('logp',data=np.array(self.logp))
         f.close()
         
+    def parse_samples(self,burn_in_frac = 0.5, thinning = None):
+        '''
+        This methods loads a .h5 file with the MCMC samples 
+        of EoS hyper-parameters and parses them for use by the 
+        plotting functions.
+        see https://emcee.readthedocs.io/en/stable/tutorials/autocorr/
+        for some documentation on choosing thinning and burn-in
+        
+        burn_in_frac  :: fraction of samples to discard from each chain
+                         for MCMC burn in. Default corresponds to discarding 
+                         half the samples in each chain.
+        
+        thinning      :: Number of samples to skip in each chain post
+                         burn in. "None" implements the default value 
+                         which is either (length of chain)/50 or half of
+                         the maximum integrated autocorrelation time. The 
+                         former is used in case autocorrelation analysis 
+                         yields throws non-convergence error. For no thinning 
+                         set thinning=1
+        
+        '''
+        
+        Ns=self.samples.shape
+        burn_in=int(Ns[0]*burn_in_frac)
+        samples=[]
+        
+        if thinning is None:
+            thinning=int(Ns[0]/50.)
 
+            try:
+                thinning=int(max(mc.autocorr.integrated_time(self.samples))/2.)
+            except mc.autocorr.AutocorrError as e:
+                print(e)
+        
+        for i in range(burn_in,Ns[0],thinning):
+            for j in range(Ns[1]):
+                samples.append(self.samples[i,j,:])
+
+        return np.array(samples)
+    
+    def load_samples(self,filename):
+        '''
+        If the pltotting function is called in post-
+        processing i.e. as part of a different script 
+        than the one that ran the sampling, then this 
+        function can be called to load the EoS hyper-parameter
+        samples.
+        
+        filename :: h5py file containing MCMC samples of 
+                    EoS hyper-parameters
+        '''
+        
+        with h5py.File(filename,'r') as f:
+            self.samples = np.array(f['chains'])
+            self.logp = np.array(f['logp'])
```

Comment:
When is this actually used?

Reply:
As mentioned in the doc string, this is used in case there is a need to reproduce plots after the samples have been generated and saved. For example, if the program that calls and runs the sampler terminates after saving the samples then this function would be needed. This was added upon the reviewer's recommendation.

Reply:
Is there any reason to prefer HDF5 format for saving the posteriors? It is just a simple Nx4 array that we are saving, right? We could just save it in JSON or even plain text format. I always find HDF5 a bit opaque because I cannot just open it on vim and see the content.

Reply:
Its a (N x nwalkers x 4) array for samples. We also save the (N x nwalkers) logp's (which might not be necessary at the moment, but can be used for computing sampling statistics if anyone wants to. I have seen that standard pe/sampling convention is to save logp as well). Do we gain anything from changing to json? It will entail multiple code changes in other other existing functions as well.

Reply:
I personally find h5 easier to deal with because I can directly load multi-dimensional arrays with one line but that might just be due to my lack of knowledge regarding how to do the same with json if it is doable. I get your point though. However, given the code change issue and the fact that ligo production PE's are stored in hdf5, I guess we are keeping this as is for now? Should I mark this as resolved then?

Reply:
Yes, sure. That is fine.